sql/tsql/TSqlParser.g4

```diff
@@ -4115,6 +4220,43 @@ xml_data_type_methods
     | modify_method
     ;
 
+// https://learn.microsoft.com/en-us/sql/t-sql/functions/date-bucket-transact-sql?view=sql-server-ver16
+dateparts_9
+    : (YEAR | YEAR_ABBR)
+    | (QUARTER | QUARTER_ABBR)
+    | (MONTH | MONTH_ABBR)
+    | (DAY | DAY_ABBR)
+    | (WEEK | WEEK_ABBR)
+    | (HOUR | HOUR_ABBR)
+    | (MINUTE | MINUTE_ABBR)
+    | (SECOND | SECOND_ABBR)
+    | (MILLISECOND | MILLISECOND_ABBR)
```

Comment:
Parentheses are excess:

```suggestion
    : YEAR | YEAR_ABBR
    | QUARTER | QUARTER_ABBR
    | MONTH | MONTH_ABBR
    | DAY | DAY_ABBR
    | WEEK | WEEK_ABBR
    | HOUR | HOUR_ABBR
    | MINUTE | MINUTE_ABBR
    | SECOND | SECOND_ABBR
    | MILLISECOND | MILLISECOND_ABBR
```

Reply:
Sure, I wanted to highlight datepart with their abbreviations but to lightweight parser it's really worth to remove the parentheses.